File: wallet_syncronizer/pkg/service/goplus/goplus.go
package goplus

import (
	"errors"
	"github.com/GoPlusSecurity/goplus-sdk-go/api/token"
	"github.com/GoPlusSecurity/goplus-sdk-go/pkg/errorcode"
	"github.com/GoPlusSecurity/goplus-sdk-go/pkg/gen/models"
)

type Service struct {
	tokenAddress string
}

func NewService(tokenAddress string) *Service {
	return &Service{tokenAddress: tokenAddress}
}

func (s *Service) ScamCheck() (response models.ResponseWrapperTokenSecurityResultAnon, err error) {
	tokenSecurity := token.NewTokenSecurity(nil)
	chainId := "1"
	contractAddresses := []string{s.tokenAddress}
	data, err := tokenSecurity.Run(chainId, contractAddresses)
	if err != nil {
		return response, err
	}
	if data.Payload.Code != errorcode.SUCCESS {
		return response, err
	}
	var ok bool
	response, ok = data.Payload.Result[s.tokenAddress]

	if ok {
		return response, nil
	} else {
		return response, errors.New("result not contains token address")
	}
}

File: wallet_syncronizer_node/service/goplus.js
const { GoPlus} = require('goplus-sdk-js');

module.exports = (logger) => {
    const scamCheck = async (tokenAddress) => {
            try {
                const chainId = '1';
                const contractAddresses = [tokenAddress];

                const data = await GoPlus.tokenSecurity(chainId, [contractAddresses], 30);

                if (data.payload.code !== 'SUCCESS') {
                    logger.error(`ScamCheck failed with code: ${data.payload.code}`);
                    return null;
                }

                const tokenResponse = data.payload.result[tokenAddress];

                if (tokenResponse) {
                    return tokenResponse;
                } else {
                    logger.error('Result does not contain token address');
                    return null;
                }
            } catch (error) {
                logger.error('ScamCheck failed:', error);
                return null;
            }

    };

    return {
        scamCheck,
    };
};

